### PR TITLE
operator: Allow endpoint_suffix in azure object storage secret

### PR DESCRIPTION
**Pull request by periklis**
**What this PR does / why we need it**:
Enable using LokiStack with Azure Blob Storage behind a private endpoint.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: operator/internal/handlers/internal/storage/secrets.go [hunk 1]
```diff
@@ -56,11 +56,15 @@ func extractAzureConfigSecret(s *corev1.Secret) (*storage.AzureStorageConfig, er
 		return nil, kverrors.New("missing secret field", "field", "account_key")
 	}
 
+	// Extract and validate optional fields
+	endpointSuffix := s.Data["endpoint_suffix"]
```
Comment (JoaoBraveCoding) [inline operator/internal/handlers/internal/storage/secrets.go:60]: I'm a bit confused by the comment, we are only extracting the field correct?

**Review by JoaoBraveCoding — COMMENTED** [review]
Just one minor comment, still have to test

File: operator/internal/handlers/internal/storage/secrets.go (hunk 1 above)
Comment (periklis) [reply operator/internal/handlers/internal/storage/secrets.go:60]: It's just a hint to make the section more human readable. I imagine that we extra more optional fields if requested.

File: operator/internal/manifests/internal/config/loki-config.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ common:
       container_name: {{ .Container }}
       account_name: {{ .AccountName }}
       account_key: {{ .AccountKey }}
+      endpoint_suffix: {{ .EndpointSuffix }}
```
Comment (xperimental) [inline operator/internal/manifests/internal/config/loki-config.yaml:17]: Should we make this optional in the output as well?

```suggestion
      {{- with .EndpointSuffix }}
      endpoint_suffix: {{ . }}
      {{- end }}
```

**Review by xperimental — COMMENTED** [review]
I tested this on a cluster by providing bogus secret values and checking the generated config. It seems fine, just one small comment.

**Review by xperimental — APPROVED** [review]
Generation works, but I've not tested with any "real" Azure bucket.

**Pull request merged**